### DISCUSSION TITLE
How to change document text encoding or specify encoding of rendered text

### DISCUSSION THREAD

**Discussion by PetrKaleta** [gettalong/hexapdf#239] [Q&A] [answered]
Hello,
I'd like to use `hexapdf` to modify existing PDF document template, which was created in Adobe InDesign on Windows and contains central european characters. So its most probably in Windows 1250 or ISO 8859-2 encoding.

When i add that contains characters with diacritics, its not rendered properly or some of them are missing.

Here is the sample code i use:

```ruby
HexaPDF::Document.open(TEMPLATE_PATH).tap do |doc|
  page = doc.pages[0]

  canvas = page.canvas(type: :overlay)
  canvas.font('Helvetica', size: 10)

  canvas.text('ěščřžýáíé', at: [0, 0])

  doc.write('foo.pdf', optimize: true)
end
```

When I try to use `canvas.font('Helvetica', size: 10, custom_encoding: true)` it seems to start working when I open rendered PDF on OSX or iOS, but its not on Android (some characters are missing). 

Any idea how to fix that?

**Comment by gettalong** [comment, marked as answer]
So, the first error you get `Type1 encoding has no codepoint for ecaron (HexaPDF::Error)` is expected since the default encoding used by HexaPDF (which is **not** related to any encoding used in an opened PDF) just doesn't contain that codepoint.

As you found out you can work around this by using the `custom_encoding: true` option. When I reproduce this I get the following result (in various viewers like okular, evince and Adobe Reader)

![image](https://user-images.githubusercontent.com/4104/226659768-dc56f1ab-55e6-46d3-a8da-8bfee26d66ba.png)

It also works with mupdf on Android. So my guess is that the viewer you are using on Android can't handle non-standard Type1 encodings.

What you can try is using a TrueType font instead. This will also allow you to display much more characters.

**Comment by PetrKaleta** [comment]
Than you, using a TrueType font fixed this issue!
